### PR TITLE
Enhance debugger

### PR DESCRIPTION
**Pull request by qinweigong**
@sojanpr @aaustin

### CODE REVIEW

**Comment by qinweigong** [comment]
Yah, I actually thought of having only a singleton BranchWindowCallback. The thing is it needs the current activity to work, which means when switching activities, developers need to set the target activity each time?

**Comment by qinweigong** [comment]
Good catch!

Taking notes of our discussion:
Whenever a dialog is loaded, its onStart() is called, hence initSession is called, so debugger listener is attached to the dialog.
However, when the dialog goes away, the activity under it won't have its onStart() called, so the listener is still with the (now gone) dialog! This might well be the reason for those support requests complaining the debugger is not working!

With the new session management, we can attach the listener back to the activity after the dialog above it goes away by overriding `onActivityResumed`. Even better, we discovered a better (more sensitive) way - WindowCallback.

File: Branch-SDK/src/io/branch/referral/Branch.java [hunk 1]
```diff
@@ -935,10 +944,9 @@ public boolean onTouch(View v, MotionEvent ev) {
 					return true;
```
Comment (qinweigong) [inline Branch-SDK/src/io/branch/referral/Branch.java]: @sojanpr are you talking about this line? so we should return false here?

**Comment by sojanpr** [comment]
cool
:+1:

**Pull request merged**